paimon-common/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -3227,6 +3271,56 @@ public String toString() {
         }
     }
 
+    /** Actions performed during table writing. */
+    public enum WriteAction {
+
+        // All write actions will be performed, this is the default behavior.
+        ALL("all"),
+
+        // Actions during commit.
+        PARTITION_EXPIRE("partition-expire"),
+        SNAPSHOT_EXPIRE("snapshot-expire"),
+        TAG_AUTOMATIC_CREATION("tag-automatic-creation"),
+
+        // Actions during writing.
+        FULL_COMPACT("full-compact");
```

Comment:
I cannot get why you want to add this. Full compact should be separate option.

Reply:
Should I add parameter `write-action.compact.enable` separately to control whether to execute compaction?
In my original conception, `compact` is one of the `write-actions` types like the others.

Reply:
Is it more reasonable to directly use "compact"? 

From the implementation code, this parameter only controls the compact behavior, while whether to perform a full compact or a minor compact is controlled by other parameters.